Add documentation about sync-loop (#74)
kyma-project/rt-bootstrapper: docs/contributor/resource-synchronisation.md

The following described behavior represents an interim solution. A long‑term architecture is planned in which the controller loop will directly synchronize configuration data to the runtimes without relying on indirect signaling.

## Components

### Controller Loop

A custom Kubernetes controller responsible for observing changes in selected cluster resources on KCP and initiating downstream actions. The watched resources are copied to  Kyma runtimes. Each resource change has to be synchronized from KCP to Kyma runtimes.

### Watched Resources

The controller loop monitors the following Kubernetes objects:

| Resource Type | Purpose | 
| --- | --- | 
| **Pull Secret** | Provides authentication credentials for pulling container images form private registries. | 
| **ClusterTrustBundle** | Supplies trust anchors (e.g., CA certificates) required by runtimes which interact with the BTP backend services. | 
| **Webhook ConfigMap** | Contains configuration for the Runtime Bootstrapper webhook. | 

> ### Runtime Custom Resource (RuntimeCR)

Author: ```suggestion
### Runtime Custom Resource
```

> A custom resource representing a managed runtime instance.

Reviewer: ```suggestion
A custom resource (CR) representing a managed runtime instance.
```

1. Identifies all affected `Runtime` CR objects.
2. Applies or updates a specific label (for example, [`operator.kyma-project.io/force-patch-reconciliation=true`](https://github.com/kyma-project/kyma-infrastructure-manager/blob/c1d2f48a9b446b3374528278b46ea9be23ff622a/pkg/reconciler/annotations_utils.go#L4C32-L4C83)) on each `Runtime` CR.
3. The Kyma Infrastructure Manager component observes this label change.
4. The Kyma Infrastructure Manager reconciles the corresponding runtimes to ensure they receive the updated configuration.

This mechanism uses the `Runtime` CR label as a signaling channel between the controller loop and the Kyma Infrastructure Manager.

### Rationale for the Interim Approach

The labeling strategy provides a lightweight and low‑risk integration path with the following advantages:

* No direct modification of runtime resources is required.
* Existing reconciliation logic in the Kyma Infrastructure Manager remains unchanged.
* The controller loop only signals intent rather than performing the full synchronization.

This allows incremental rollout and testing of the controller loop without impacting runtime stability.

## Long‑Term Architecture

> In the future design, the controller loop will no longer rely on labeling `Runtime` CR objects to trigger reconciliation. Instead, it will directly propagate configuration changes to the runtimes without any Kyma Infrastructure involvement.

Reviewer: ```suggestion
In the future design, the controller loop will no longer rely on labeling `Runtime` CR objects to trigger reconciliation. Instead, it will directly propagate configuration changes to the runtimes without KIM involvement.
```
